src/sessions.ts

```diff
@@ -610,65 +608,48 @@ function attemptTransaction<T>(
   }
 
   if (!isPromiseLike(promise)) {
-    session.abortTransaction().catch(() => null);
-    return Promise.reject(
-      new MongoInvalidArgumentError('Function provided to `withTransaction` must return a Promise')
+    await session.abortTransaction().catch(() => null);
```

Comment:
```suggestion
    try {
      await session.abortTransaction();
    } catch (error) {
      squashError(error);
    }
```

Can you remove the `/* eslint-disable github/no-then */` at the top of the file and align this bit with the rest of the way we squash errors?